src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs

```diff
@@ -81,6 +81,58 @@ public static ImmutableArray<T> Create<T>(T item1, T item2, T item3, T item4)
             return new ImmutableArray<T>(array);
         }
 
+        /// <summary>
+        /// Creates an <see cref="ImmutableArray{T}"/> with the specified elements.
+        /// </summary>
+        /// <typeparam name="T">The type of element stored in the array.</typeparam>
+        /// <param name="items">The elements to store in the array.</param>
+        /// <returns>An immutable array.</returns>
+        public static ImmutableArray<T> Create<T>(ReadOnlySpan<T> items)
+        {
+            if (items.Length == 0)
+            {
+                return ImmutableArray<T>.Empty;
+            }
+
+            T[] array = items.ToArray();
+            return new ImmutableArray<T>(array);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="ImmutableArray{T}"/> with the specified elements.
+        /// </summary>
+        /// <typeparam name="T">The type of element stored in the array.</typeparam>
+        /// <param name="items">The elements to store in the array.</param>
+        /// <returns>An immutable array.</returns>
+        public static ImmutableArray<T> Create<T>(Span<T> items)
+        {
+            return Create((ReadOnlySpan<T>)items);
+        }
+
+        /// <summary>
+        /// Produce an immutable array of contents from specified elements.
+        /// </summary>
+        /// <typeparam name="T">The type of element in the list.</typeparam>
+        /// <param name="items">The elements to store in the array.</param>
+        /// <returns>An immutable array.</returns>
+        public static ImmutableArray<T> ToImmutableArray<T>(this ReadOnlySpan<T> items)
+        {
```

Comment:
The extension method is not checking if `items` is empty, however the static `Create` method as defined above does do so. I'm not sure which approach is the best (running benchmarks might help) but shouldn't the two be consistent?

Reply:
Missed "fast return" for this method previously. Changed to be consistent with other methods.

Reply:
I ignored [ReadOnlySpan<T>.ToArray()](https://github.com/dotnet/runtime/blob/eeaee44e6fe4425db044967e025255e6a12c97a9/src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs#L371) itself also checks if items is empty. If yes, it just returns Array.Empty(). Consider [ImmutableArray<T>.Empty implementation](https://github.com/dotnet/runtime/blob/eeaee44e6fe4425db044967e025255e6a12c97a9/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Minimal.cs#L44), so:

- empty case: same checking count (1), except that one use 'new T[0]', another use 'Array.Empty()' in the end
- non-empty case: one has 1 length check, another has 2 length check

I will try to use "dotnet/performance" repo to run benchmark. Is this correct way ?

Reply:
Yes dotnet/performance is the best way to do it, let me know you if you need help setting it up.

Reply:
Finally, succeed in making dotnet/perf happy to run benchmark against my runtime artifact.. Result is: seems pre checking empty has a little shorter time.
![image](https://user-images.githubusercontent.com/72254037/151695638-9df526f3-1eff-4eb1-96ea-853df110c359.png)

I added 2 ImmutableArray class methods (`CreateByPreCheckEmpty()`, `CreateEmptyByDirectToArray()`) for 2 type of implementations to be compared and run several time and got same trend. Benchmark test code.:
```
    public class ImmutableArrayTest
    {
        private int[] _arrayData;

        [GlobalSetup]
        public void Setup()
        {
            _arrayData = new int[100];
        }

        [Benchmark]
        public ImmutableArray<int> CreateEmptyByPreCheckEmpty() => ImmutableArray.CreateByPreCheckEmpty(ReadOnlySpan<int>.Empty);

        [Benchmark]
        public ImmutableArray<int> CreateEmptyByDirectToArray() => ImmutableArray.CreateByDirectToArray(ReadOnlySpan<int>.Empty);

        [Benchmark]
        public ImmutableArray<int> CreateArrayByPreCheckEmpty() => ImmutableArray.CreateByPreCheckEmpty(new ReadOnlySpan<int>(_arrayData));

        [Benchmark]
        public ImmutableArray<int> CreateArrayByDirectToArray() => ImmutableArray.CreateByDirectToArray(new ReadOnlySpan<int>(_arrayData));
    }
```

So let us use current ToImmutableArray() and Create() implementation.